client/src/pages/PostSolution.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState, ChangeEvent, FormEvent } from "react";
+import { useParams } from "react-router-dom";
+import { Col, Container, Form, Row } from "react-bootstrap";
+import Button from "../components/Button.tsx";
+
+interface FormData {
+  title: string;
+  description: string;
+  diagram: File | null;
+}
+
+// TODO: Add form validation
+// TODO: Include data about the challenge like title
+
+const PostSolution: React.FC = () => {
+  const { challengeId } = useParams();
+  const [formData, setFormData] = useState<FormData>({
+    title: "",
+    description: "",
+    diagram: null,
+  });
+
+  const handleChange = (
+    e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ) => {
+    const { name, value } = e.target;
+    setFormData((prevData) => ({
+      ...prevData,
+      [name]: value,
+    }));
+  };
+
+  const handleFileChange = (e: ChangeEvent<HTMLInputElement>) => {
+    if (e.target.files && e.target.files.length > 0) {
+      const file = e.target.files[0];
+      setFormData((prevData) => ({
+        ...prevData,
+        diagram: file,
+      }));
+    }
+  };
+
+  const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    // Handle form submission here
+    console.log(formData);
+    console.log(challengeId);
+  };
+  return (
+    <Container>
+      <Row className={"my-5 justify-content-center"}>
+        <Col md={4} className={"bg-gray-200 rounded py-5 px-md-5"}>
+          <h2>Submit Your Solution!</h2>
+          <Form onSubmit={handleSubmit}>
+            <Form.Group className="mb-3" controlId="title">
+              <Form.Label>Title</Form.Label>
+              <Form.Control
+                type="text"
+                name="title"
+                value={formData.title}
+                onChange={handleChange}
+                required
+              />
+            </Form.Group>
+            <Form.Group className="mb-3" controlId="description">
+              <Form.Label>Description</Form.Label>
+              <Form.Control
+                as="textarea"
+                name="description"
+                value={formData.description}
+                onChange={handleChange}
+                rows={4}
+                required
+              />
+            </Form.Group>
+            <Form.Group className="mb-3" controlId="diagram">
+              <Form.Label>Diagram</Form.Label>
+              <Form.Control
+                type="file"
+                name="diagram"
+                onChange={handleFileChange}
+                accept="image/*"
```

Comment:
I think the diagram should be `required` as well.

Reply:
We cannot store the diagram on the database. We need to have a storage service because an image file is too large to load onto one post request.

Reply:
What file types are we allowing users to submit?

Reply:
I am currently storing svg in the database while working on the editor. We can discuss file formats.

Reply:
any image format would do, including svg